File: go/125ValidPalindrome.go
package main

import (
	"fmt"
	"strings"
	"unicode"
)

func isAlphanumeric(char rune) bool {
	return unicode.IsDigit(char) || unicode.IsLetter(char)
	// charAsciNum := int(char)
	// return ((charAsciNum >= int('a') && charAsciNum <= int('z')) ||
	// 	(charAsciNum >= int('A') && charAsciNum <= int('Z')) ||
	// 	(charAsciNum >= int('0') && charAsciNum <= int('9')))
}

func isPalindrome(s string) bool {
	lowerS := strings.ToLower(s)
	runeS := []rune(lowerS)
	fmt.Println(len(runeS))
	for i, j := 0, len(runeS)-1; i < j; i, j = i+1, j-1 {
		for i < j && !isAlphanumeric(runeS[i]) {
			fmt.Printf("%c\n", runeS[i])
			i++
		}
		for i < j && !isAlphanumeric(runeS[j]) {
			fmt.Printf("%c\n", runeS[j])
			j--
		}
		fmt.Println(runeS[i], runeS[j])
		if runeS[i] != runeS[j] {
			return false
		}
	}
	return true
}

func main() {
	// fmt.Println(isPalindrome("race a car"))
	fmt.Println(isAlphanumeric('a'))
}

File: php/125ValidPalindrome.php
<?php
function isAlphanumeric(string $char):bool {
  $charAsciNum = ord( $char);
  return (
    ($charAsciNum >= ord('a') && $charAsciNum <= ord('z'))||
    ($charAsciNum >= ord('A') && $charAsciNum <= ord('Z'))||
    ($charAsciNum >= ord('0') && $charAsciNum <= ord('9'))
  );
}

function isPalindrome(string $str):bool {
  for($i = 0,$j = strlen($str)-1; $i < $j; $i++,$j--) {
    while(!isAlphanumeric($str[$i]) && $i < $j)
      $i++;
    while(!isAlphanumeric($str[$j]) && $i < $j)
      $j--;
    if(strtolower($str[$i]) !== strtolower($str[$j]))
      return false;
  }   
  return true;
}